File: src/main/java/io/saso/dash/services/DBEntityProvider.java
package io.saso.dash.services;

import io.saso.dash.database.entities.*;

public interface DBEntityProvider
{
    User user() throws Exception;

    Provider provider(Service service) throws Exception;

    ProviderUser providerUser(Service service) throws Exception;

    AuthToken authToken(Service service) throws Exception;
}

File: src/main/kotlin/io/saso/dash/services/DBEntityProvider.kt
package io.saso.dash.services

import io.saso.dash.database.entities.*

public interface DBEntityProviderOLD
{
    fun liveToken(): DBLiveToken

    fun user(): DBUser

    fun provider(service: String): DBProvider

    fun providerUser(service: String): DBProviderUser

    fun authToken(service: String): DBAuthToken
}
